src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +185,98 @@ public void Initialize(TelemetryConfiguration configuration)
                             portalSender.Send(traceEvent);
                         }
 
+                        // set up heartbeat
+                        this.HeartbeatProvider.Initialize(configuration);
+
                         this.isInitialized = true;
                     }
                 }
             }
         }
-        
+
+        /// <summary>
+        /// Add a new Health Heartbeat property to the payload sent with each heartbeat.
+        /// 
+        /// To update the value of the property you are adding to the health heartbeat, 
+        /// <see cref="DiagnosticsTelemetryModule.SetHealthProperty"/>.
+        /// 
+        /// Note that you cannot add a HealthHeartbeatProperty with a name that already exists in the HealthHeartbeat
+        /// payload, including (but not limited to) the name of SDK-default items.
+        /// 
+        /// </summary>
+        /// <param name="propertyName">Name of the health heartbeat value to add</param>
+        /// <param name="propertyValue">Current value of the health heartbeat value to add</param>
+        /// <param name="isHealthy">Flag indicating whether or not the property represents a healthy value</param>
+        /// <returns>True if the new payload item is successfully added, false otherwise.</returns>
+        public bool AddHealthProperty(string propertyName, string propertyValue, bool isHealthy)
+        {
+            try
+            {
+                return this.HeartbeatProvider.AddHealthProperty(propertyName, propertyValue, isHealthy);
+            }
+            catch (Exception e)
+            {
+                CoreEventSource.Log.LogError("Could not add heartbeat property. Exception: " + e.ToInvariantString());
```

Comment:
try to avoid string formatting when using event source

Reply:
Yes, done. @cijothomas informed me about the CoreEventSource logging pattern we are to use, and I have added to that class where I thought it appropriate.